### ISSUE TITLE
About specificity_at_sensitivity metric

### ISSUE THREAD

**Issue by oguz-hanoglu** [Lightning-AI/torchmetrics#2189]
## 🚀 Feature

The metric [SpecificityatSensitivity](https://torchmetrics.readthedocs.io/en/stable/classification/specificity_at_sensitivity.html#binaryspecificityatsensitivity) returns Tuple.

Is it aligned with the general structure of Torchmetrics? Generally, single metric is returned.

**Comment by ozen** [comment]
I think returning tuple breaks lightning integration, since it expects a single metric value to be returned.

Can we introduce a `return_thresholds` parameter to the constructor of the metric to control this behavior?

**Comment by oguz-hanoglu** [comment]
@awaelchli We need your experience here :) Do you have an idea or do you know somebody who can have an idea on the general structure of torch metrics.

@ozen Seems like a good idea.

**Comment by awaelchli** [comment]
@oguz-hanoglu @ozen I assume you are talking about the `self.log` feature in Lightning, that can accept a metric object as input, and will evaluate it automatically?

I think there will always be some "metrics" that produce a non-scalar. For example, the [confusion matrix](https://torchmetrics.readthedocs.io/en/stable/classification/confusion_matrix.html#id3) is a very useful "metric" in torchmetrics and it outputs a NxN tensor. So in regards to torchmetrics yes, I think it is generally acceptable to have them return non-scalars. 

For Lightning, we don't have a good way to handle this right now and it is not in scope at the moment. We would need to brainstorm this on a feature request and see if there is enough demand for this.

The user today shouldn't be blocked by this though. It's always possible to evaluate the metric separately and either unpack the tuple and then log whatever you want with `self.log()`, or further process the non-scalar output and visualize it in a different way. 

These are my quick thoughts :)

**Comment by awaelchli** [comment]
Whether it makes sense to add an argument to `BinarySpecificityAtSensitivity` like @ozen, I have no objections but leave it up to torchmetrics team to decide :))

**Comment by ozen** [comment]
@awaelchli thanks for the clarification.

Computing `SpecificityAtSensitivity` at batch level doesn't make sense, and unnecessarily adds the cost of computing the ROC curve.

Also, it doesn't perform a reduction like mean or sum.

So a possible workaround, I think, is to use `metric.update()` and not to use `self.log()` in the step method, and to use `metric.compute()` and `metric.reset()` in the epoch end hook. 

So, basically, we cannot use the torchmetrics feature of `self.log()`.
